### ISSUE TITLE
Expose extensions in the ConfigDump endpoint.

### ISSUE THREAD

**Issue by jpeach** [envoyproxy/envoy#9097]
*Title*: 

Expose extensions in the ConfigDump endpoint.

*Description*:

The set of extensions in the envoy binary are logged at startup, but not otherwise exposed. If these extensions were exposed by the [config_dump](https://www.envoyproxy.io/docs/envoy/latest/api-v2/admin/v2alpha/config_dump.proto) endpoint, then it would be easier for operational tooling (e.g facter) to track what extensions are available. Another use that I have in mind for this is to let Kubernetes ingress controllers inspect an envoy process to ensure that the features it needs are compiled in.

**Comment by junr03** [comment]
This would be neat! I wonder if we should put it in the config_dump endpoint or expose it in a separate endpoint.

**Comment by mattklein123** [comment]
> This would be neat! I wonder if we should put it in the config_dump endpoint or expose it in a separate endpoint.

+1 on this feature. I don't feel super strongly about whether it's part of config_dump or not. I guess I would mildly favor config_dump for simplicity.

**Comment by jpeach** [comment]
@mattklein123 @junr03 Could you please assign this issue to me?

Adding this to `config_dump` looks like this:
```json
    {
      "@type": "type.googleapis.com/envoy.admin.v2alpha.ExtensionsConfigDump",
      "categories": {
        "health_checkers": {
          "extensions": [
            "envoy.health_checkers.redis"
          ]
        },
        "filters.http": {
          "extensions": [
            "envoy.buffer",
            "envoy.cors",
            "envoy.csrf",
            "envoy.ext_authz",
            "envoy.fault",
            ...
          ]
        },
    }    
```

**Comment by lizan** [comment]
(same as I commented in PR): extensions will likely end up in Node as part of client capabilities that we discussed in #8332, so I don't think we should put in a separate config_dump proto?

**Comment by jpeach** [comment]
@lizan Yep that makes sense. WDYT about adding the Node port message to the server_info admin endpoint? That seems more consistent to me.

**Comment by jpeach** [comment]
@yanavlasov @lizan 

I started looking again at adding the Node message to the server info admin endpoint. There's a few things in the Node message that probably only make sense for a xDS server (e.g. client_features, user_agent_version), but other fields seem generally useful to systems that might poll the server_info (e.g. locality, metadata, extensions). So it seems like a bit of an uncomfortable fit in the server-info context.

I think that if I was operating Envoy with Puppet that I might find the extensions useful to have in server_info, but my personal need for this information is satisfied through xDS.

So there's 3 options:

1. Just drop it, since there's no strong use case
1. Copy only the relevant Node fields to ServerInfo
1. Copy the whole Node and don't worry about the xDS-specific fields

Lemme know what you think, I'm happy to continue working on one of these options :)

**Comment by yanavlasov** [comment]
I think admin interface as it is now is rather problematic from security standpoint. It is plaintext and unauthenticated. I would rather not extend it unless there is immediate business need.
I think there is a need for admin endpoint that is configured as a filter chain and where requests can use secure transport and be authenticated.
